### PR TITLE
Implement TestExtract for DATE type

### PR DESCRIPTION
**Pull request by choiwaiyiu**
Fixes https://github.com/prestosql/presto/issues/5626

### CODE REVIEW

**Comment by martint** [comment]
Can you fix the build issues and rebase?

```
Error:  Failed to execute goal com.mycila:license-maven-plugin:2.3:check (default) on project presto-main: Some files do not have the expected license header -> [Help 1]
```

**Comment by choiwaiyiu** [comment]
@martint The following tests still pass after rebase. Is it expected?

- `assertThat(assertions.expression("hour(DATE '2020-05-10')")).matches("BIGINT '0'");`
- `assertThat(assertions.expression("minute(DATE '2020-05-10')")).matches("BIGINT '0'");`
- `assertThat(assertions.expression("second(DATE '2020-05-10')")).matches("BIGINT '0'");`
- `assertThat(assertions.expression("timezone_hour(DATE '2020-05-10')")).matches("BIGINT '13'");`
- `assertThat(assertions.expression("timezone_minute(DATE '2020-05-10')")).matches("BIGINT '0'");`

**Comment by martint** [comment]
Yes, it’s unfortunate, but those work because of implicit coercion’s between those types and timestamp with time zone.

**Comment by choiwaiyiu** [comment]
I see. Thank you for the confirmation.

File: presto-main/src/test/java/io/prestosql/operator/scalar/date/TestExtract.java [hunk 1]
```diff
@@ -42,20 +42,23 @@ public void teardown()
     @Test
     public void testYear()
     {
+        assertThat(assertions.expression("year(DATE '2020-05-10')")).matches("BIGINT '2020'");
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/operator/scalar/date/TestExtract.java:45]: In other TestExtract test classes we put `EXTRACT(...` cases first within a test method, then followed by the cases covering the convenience methods. See `io.prestosql.operator.scalar.timestamptz.TestExtract` for a "canonical example". Can you rearrange the cases here?

**Comment by choiwaiyiu** [comment]
@findepi Should I rebase and use `AbstractTestExtract`?

**Comment by findepi** [comment]
@choiwaiyiu  Yes, please do.

**Comment by findepi** [comment]
Merged, thanks!

**Pull request closed without merging**